### ISSUE TITLE
Training fails with "CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure" on Windows 10

### ISSUE THREAD

**Issue by dgrechka** [tensorflow/tensorflow#39958]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): **yes**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Windows 10**
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: **No**
- TensorFlow installed from (source or binary): **binary (via pip install)**
- TensorFlow version (use command below): **v2.2.0-rc4-8-g2b96f3662b 2.2.0**
- Python version: **3.6.8**
- Bazel version (if compiling from source): Not applicable
- GCC/Compiler version (if compiling from source): Not applicable
- CUDA/cuDNN version: **cuda 10.1.243_426.00_win10 (CUDA Toolkit 10.1 update2 Archive ) / cuDNN v7.6.5 (November 5th, 2019), for CUDA 10.1**
- GPU model and memory: **GeForce GTX 1070, Compute Capability 6.1, 8Gb; 
also on GeForce GTX 1050 Ti 4Gb on another Win 10 machine with the same cuda version installed**

**Describe the current behavior**
I use TF Keras API to define the model and the training process.

The training stops with the error:
```
2020-05-28 18:52:43.846256: E tensorflow/stream_executor/cuda/cuda_event.cc:29] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure
2020-05-28 18:52:43.846616: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:273] Unexpected Event status: 1

exit code -1073740791
```

after several training epochs.

It happens in a reproducible way (the same python file execution fails after the same number of epochs).
When I change the random seed (thus the dataset is shuffled differently) the process fails after different number of epochs!
Changing batch size also effects the number of epochs until the training fail (randomly)!
I use batch size of 1 or 2 as I have rather large model.

**Describe the expected behaviour**
Training runs without the error requested number of epochs.

**Comment by amahendrakar** [comment]
@dgrechka,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here.

Also, please check [this](https://github.com/tensorflow/tensorflow/issues/37942) similar issue and let us know if it helps. Thanks!

**Comment by dgrechka** [comment]
Finally I've separated out the repro out of my project.

It seems that it fails on considerably large number of training samples.
If I try to reproduce it with smaller dataset, the issue is not triggered.

### How to reproduce

**On windows 10**  (see environment details in the very first post).

1. Download the [zip archive](https://grechka.family/dmitry/sandbox/issue_39958_repro.zip) (quite large ~871Mb, but smaller samples count does not trigger the issue.)
2. extract it
3. install python dependencies with `pip install -r .\pip_deps.txt`
4. run the repro script with `python .\run_repro.py`

The file `stdout_stderr_showing_the_issue.log` in the archive contains the stdout and stderr which I get when I run this repro (including the error).

**Comment by amahendrakar** [comment]
@dgrechka,
I was able to run the code without any issues, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/2f833285e241e4c355cd2703f9e34d46/39958.ipynb).

Could you please upgrade the the stable version of TensorFlow 2.2 and check if you are facing the same issue. Thanks!

**Comment by dgrechka** [comment]
@amahendrakar , thanks!

As you can see in pip_deps.txt, I use the latest stable version of tensorflow `tensorflow==2.2.0` which is release of May 2, 2020.

This issue is related to Windows only.
There is no such issue on linux (I tried Ubuntu 18.04)

Did you run the repro on Windows?

**Comment by amahendrakar** [comment]
@dgrechka,
Thank you for the update. Currently, I do not have a Windows machine to test the code.

**Comment by sanjoy** [comment]
Can you please run the test with the `CUDA_LAUNCH_BLOCKING` env variable set to `1`?  That will help us narrow down the issue to a specific kernel launch.

Please also upload the full log as a gist so that we don't have download the whole archive to look at the log.

**Comment by dgrechka** [comment]
The issue goes for a some time after the reboot (then comes back again). Give me some more time to reproduce with the flags requested, I'm postponed the model training on Windows. Will return and give more repro details.

**Comment by Saduf2019** [comment]
@dgrechka 
Is this still an issue.

**Comment by AaronAfewerqi** [comment]
I am also running TF 2.4.1 on Windows 10. And like you, my GPU is an Nvidia GeForce GTX 1050 Ti (with CUDA 11.0 installed). I'm training on a large number of images. When I tried to run my code on a sample of the data (containing about 3% of the entire data), it worked out fine. But when I fed all of the data to the model, it came up with this error after finishing some epochs (1, 2 or 3 epochs):

CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure, Unexpected Event status: 1

I wonder if you have solved your problem and if so, how. Thanks!
